ivy/functional/backends/numpy/experimental/manipulation.py

```diff
@@ -483,3 +487,25 @@ def column_stack(
     arrays: Sequence[np.ndarray], /, *, out: Optional[np.ndarray] = None
 ) -> np.ndarray:
     return np.column_stack(arrays)
+
+
+@with_supported_dtypes(
+    {"1.25.2 and below": ("float32", "float64", "int32", "int64")}, backend_version
+)
+def put_along_axis(
+    arr: np.ndarray,
+    indices: np.ndarray,
+    values: np.ndarray,
+    axis: int,
+    /,
+    *,
+    mode: str = None,
+    out: Optional[np.ndarray] = None,
+):
+    np.put_along_axis(arr, indices, values, axis)
+    return ivy.inplace_update(out, arr) if ivy.exists(out) else arr
+
+
+put_along_axis.partial_mixed_handler = lambda *args, mode=None, **kwargs: mode in [
+    "assign",
+]
```

Comment:
And I didn't look into the frontends at all. You'll have to run them and see if something needs to be changed. 
`ivy.put_along_axis` accumulates in case there are duplicate indices, so if a frontend is supposed to not do that, we'll have to clear up `indices` before calling the ivy function.

(For the missing modes in the torch frontend you can just add a ToDo.)

Reply:
I think torch supports all the modes natively

Reply:
Yes, I meant the `mean` and `mul` modes which will not be supported yet because they are missing from our `scatter_nd`.